sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -142,6 +142,62 @@ using IsKnownIdentityOp =
                   IsMinimumIdentityOp<T, BinaryOperation>::value ||
                   IsMaximumIdentityOp<T, BinaryOperation>::value>;
 
+template <typename BinaryOperation, typename AccumulatorT>
+struct has_known_identity_impl {
+  static constexpr bool value =
+      IsKnownIdentityOp<AccumulatorT, BinaryOperation>::value;
+};
+
+template <typename BinaryOperation, typename AccumulatorT, typename = void>
+struct known_identity_impl {};
+
+/// Returns zero as identity for ADD, OR, XOR operations.
+template <typename BinaryOperation, typename AccumulatorT>
+struct known_identity_impl<BinaryOperation, AccumulatorT,
+                           typename std::enable_if<IsZeroIdentityOp<
+                               AccumulatorT, BinaryOperation>::value>::type> {
+  static constexpr AccumulatorT value = 0;
+};
+
+/// Returns one as identify for MULTIPLY operations.
+template <typename BinaryOperation, typename AccumulatorT>
+struct known_identity_impl<BinaryOperation, AccumulatorT,
+                           typename std::enable_if<IsOneIdentityOp<
+                               AccumulatorT, BinaryOperation>::value>::type> {
+  static constexpr AccumulatorT value = 1;
+};
+
+/// Returns bit image consisting of all ones as identity for AND operations.
+template <typename BinaryOperation, typename AccumulatorT>
+struct known_identity_impl<BinaryOperation, AccumulatorT,
+                           typename std::enable_if<IsOnesIdentityOp<
+                               AccumulatorT, BinaryOperation>::value>::type> {
+  static constexpr AccumulatorT value = ~static_cast<AccumulatorT>(0);
+};
+
+/// Returns maximal possible value as identity for MIN operations.
+template <typename BinaryOperation, typename AccumulatorT>
+struct known_identity_impl<BinaryOperation, AccumulatorT,
+                           typename std::enable_if<IsMinimumIdentityOp<
+                               AccumulatorT, BinaryOperation>::value>::type> {
```

Comment:
Minor comment, definitely not a request to change.
This could be a little bit shorter (but same number of lines though):
```suggestion
                           enable_if_t<IsMinimumIdentityOp<
                               AccumulatorT, BinaryOperation>::value>> {
```

Reply:
I tried this during development and couldn't get it to work.  I chalked it up to me not understanding enough about how our implementation of `enable_if_t` differs from `std::enable_if`.  Can you compile locally if you make this change?  Maybe I'm doing something wrong.

Reply:
I checked our ``enable_if_t`` implementation. It is the same as in C++ standard. So, there should not be any problem. If you need help we may have a call or IM